src/clj/fluree/db/query/exec/eval.cljc

```diff
@@ -281,183 +282,277 @@
                        :status 400
                        :error  :db/invalid-query})))))
 
-(defmacro less-than
-  [var-a var-b]
-  (let [dt-a (var->dt-var var-a)
-        dt-b (var->dt-var var-b)]
-    `(neg? (compare* ~var-a ~dt-a ~var-b ~dt-b))))
-
-(defmacro less-than-or-equal
-  [var-a var-b]
-  (let [dt-a (var->dt-var var-a)
-        dt-b (var->dt-var var-b)]
-    `(or (= ~var-a ~var-b)
-         (neg? (compare* ~var-a ~dt-a ~var-b ~dt-b)))))
-
-(defmacro greater-than
-  [var-a var-b]
-  (let [dt-a (var->dt-var var-a)
-        dt-b (var->dt-var var-b)]
-    `(pos? (compare* ~var-a ~dt-a ~var-b ~dt-b))))
-
-(defmacro greater-than-or-equal
-  [var-a var-b]
-  (let [dt-a (var->dt-var var-a)
-        dt-b (var->dt-var var-b)]
-    `(or (= ~var-a ~var-b)
-         (pos? (compare* ~var-a ~dt-a ~var-b ~dt-b)))))
+(defn less-than
+  [{a :value a-dt :datatype-iri}
+   {b :value b-dt :datatype-iri}]
+  (where/->typed-val (neg? (compare* a a-dt b b-dt))))
+
+(defn less-than-or-equal
+  [{a :value a-dt :datatype-iri}
+   {b :value b-dt :datatype-iri}]
+  (where/->typed-val
+    (or (= a b)
+        (neg? (compare* a a-dt b b-dt)))))
+
+(defn greater-than
+  [{a :value a-dt :datatype-iri}
+   {b :value b-dt :datatype-iri}]
```

Comment:
Destructuring which is quite slow, and some of these functions could be hit thousands of times+ in larger query results. Because in many cases here the destructured value is only used once, I think we should prefer either `get` or could add a  `value` and `datatype` method to `TypedValue`.

Doesn't need to be done here, I think this is good for now to fix the issues... but as a note for future.